### ISSUE TITLE
Email notifier does not send an email when testing the notifier

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#12618]
**Rancher versions:**
rancher/server: v2.0.0-beta2

**Steps to Reproduce:**

1. Create cluster
1. use Add Notifier via Tools -> Notifiers
1. Choose Email, fill in the required fields
1. Hit Test, UI shows `Please check your chosen application for the notifier`. 

**Results:**
- No email is sent

The captured API call is (stripped):
```
'{"type":"notifier","name":"testemail","description":null,"clusterId":"cluster-dqdxx","smtpConfig":{"defaultRecipient":"xxx","host":"xxx","password":"xxx","port":25,"tls":false,"type":"/v3/schemas/smtpConfig","username":"xx"},"id":"cluster-dqdxx:notifier-lff2z","uuid":"2eae6926-3d01-11e8-905b-0242ac110002"}'
```

If I include `"message":"test message"`, I do receive the email:

```
'{"type":"notifier","name":"testemail","description":null,"clusterId":"cluster-dqdxx","smtpConfig":{"defaultRecipient":"xxx","host":"xxx","password":"xxx","port":25,"tls":false,"type":"/v3/schemas/smtpConfig","username":"xx"},"id":"cluster-dqdxx:notifier-lff2z","message":"test message","uuid":"2eae6926-3d01-11e8-905b-0242ac110002"}'
```

**Comment by zionwu** [comment]
@superseb The `Test` button will not send out an email.  It will just validate if the smtp config is valid or not by only calling the smtp auth API.  The reason why we don't send out an email is that the test message is usually treated as spam by email provider.

**Comment by superseb** [comment]
@zionwu Can we change the message to `Settings validated` or something, instead of `Please check your chosen application for the notifier` which is not the case for email.

**Comment by zionwu** [comment]
@superseb sure, it is a bug for the message in UI. We will have it fixed

**Comment by vincent99** [comment]
This makes no sense.. the other types send a test message, email should too.  There shouldn't be anything different about the test message versus an actual alert, and half the point of testing is to see if the message actually makes it through filtering or not.

**Comment by zionwu** [comment]
Yep, we should use a test message that is similar to an actual alert for testing email.

**Comment by tfiduccia** [comment]
Dependent on #13516

**Comment by tfiduccia** [comment]
2.0 master 5/18 - not fixed
I entered my email info and clicked on test and it gave it me a error.  I saved same notifier and it worked fine when I attached it to a alert.
![image](https://user-images.githubusercontent.com/11514927/40258306-fcd2eb50-5aa5-11e8-9c1b-751637692f9c.png)

**Comment by aiwantaozi** [comment]
@tfiduccia I could not reproduce this? Look like network problem.
